pgsqltoolsservice/connection/connection_service.py

```diff
@@ -213,16 +203,20 @@ def handle_list_databases(self, request_context: RequestContext, params: ListDat
         except ValueError as err:
             request_context.send_error(str(err))
             return
+        
         query_results = None
         try:
-            query_results = _execute_query(connection, 'SELECT datname FROM pg_database WHERE datistemplate = false;')
-        except psycopg2.ProgrammingError as err:
+            query_results = connection.list_databases()
+
+        except Exception as err:
```

Comment:
We would need to define specific error types

Reply:
I agree, but I'm not yet sure now. My idea is that we can create a helper that yields error types depending on the current provider.